File: assets/minecraft/optifine/cem/.files/update.py
from json import load, dumps

inData = load(open("models.json"))
outData = load(open("E:/Programming/GitHub/wynem/src/assets/json/cem_template_models.json"))

for entity in outData["models"]:
  if inData["models"].get(entity):
    if outData["models"][entity].get("texture_data"):
      if not isinstance(outData["models"][entity]["texture_data"], list):
        outData["models"][entity]["texture_data"] = inData["models"][entity]["texture_data"]
      else:
        print(entity)
    outData["models"][entity]["model"] = inData["models"][entity]["model"]
  else:
    print(entity)

with open("E:/Programming/GitHub/wynem/src/assets/json/cem_template_models.json", "w", newline="\n") as outFile:
  outFile.write(dumps(outData, indent = 2))

File: assets/minecraft/optifine/cem/.files/update.js
import fs from "fs"

const generated = JSON.parse(fs.readFileSync("models.json", "utf8"))
const masterPath = "E:/Programming/GitHub/wynem/src/assets/json/cem_template_models.json"
const data = JSON.parse(fs.readFileSync(masterPath, "utf8"))

let updated = 0
let missing = 0

for (const entity of Object.keys(data.models)) {
  const gen = generated.models[entity]
  if (gen) {
    delete data.models[entity].texture_data
    data.models[entity].model = gen.model
    updated++
  } else {
    console.log(`Missing from generated: ${entity}`)
    missing++
  }
}

fs.writeFileSync(masterPath, JSON.stringify(data, null, 2))
console.log(`Updated ${updated} models${missing ? `, ${missing} missing` : ""}`)
